### PR TITLE
IOS-1002 Applies `+1` prefix fix onto Banno's fork of `PhoneNumberKit`

### PR DESCRIPTION
**Pull request by bmonk**
###  References

* Fixes IOS-1002  https://banno-jha.atlassian.net/browse/IOS-1002

###  What is the goal?

* Zelle SDK would like to use `PhoneNumberKit`, but needs a fix from an unmerged PR on the upstream repo.

* Applies this `+1` prefix fix onto Banno's fork of `PhoneNumberKit.PartialFormatter`, taken from their unmerged PR https://github.com/marmelroy/PhoneNumberKit/pull/329

(also see https://github.com/marmelroy/PhoneNumberKit/pull/244/files)

That PR tests good, but the PR has been unmerged for about a year, even as other PRs merge as recently as yesterday. We've upvoted it to hopefully get it back on their radar; in the meantime, we apply their patch to our fork.

### How is it being implemented?

* Applies `PhoneNumberKit`'s prefix fix to `PartialFormatter`, directly from their unmerged PR https://github.com/marmelroy/PhoneNumberKit/pull/329

(also see https://github.com/marmelroy/PhoneNumberKit/pull/244/files)

###  Screenshots or Gifs

None.

###  How to Test

* Checkout Zelle SDK branch https://github.com/Banno/BannoZelleSDK-iOS/pull/582

* Edit the Zelle project's the `Cartfile` line 

```github "Banno/PhoneNumberKit" ~> 3.1.0```

to point to this branch:

```github "Banno/PhoneNumberKit" "bmonk/IOS-1002-applies-prefix-fix-from-unmerged-PhoneNumberKit-PR"```

* On the Zelle branch, run `carthage update --platform iOS`

* Uncomment the two Zelle tests at the bottom of `ZellePhoneNumberFormatterTests.swift` (line 20)

* Uncomment the Zelle test near the top of `TokenFormattedValueTests` (line 24)

* Select Zelle Target `BannoZelleSDK-Debug` and hit Cmd-U.

* Confirm all tests pass

###  Known Issues
None.

### CODE REVIEW

File: PhoneNumberKit/PartialFormatter.swift [hunk 1]
```diff
@@ -114,14 +114,14 @@ public final class PartialFormatter {
...
-        
+
         var finalNumber = String()
-        if self.prefixBeforeNationalNumber.count > 0 {
-            finalNumber.append(self.prefixBeforeNationalNumber)
-        }
-        if self.shouldAddSpaceAfterNationalPrefix, self.prefixBeforeNationalNumber.count > 0, self.prefixBeforeNationalNumber.last != PhoneNumberConstants.separatorBeforeNationalNumber.first {
-            finalNumber.append(PhoneNumberConstants.separatorBeforeNationalNumber)
-        }
+         if self.withPrefix, self.prefixBeforeNationalNumber.count > 0 {
```
Comment (stevekohls) [inline PhoneNumberKit/PartialFormatter.swift]: looks like some extra spaces got added here

**Review by stevekohls — CHANGES_REQUESTED** [review]
@bmonk changes look good. I still need to manually test. See my comment about spaces.

File: PhoneNumberKit/PartialFormatter.swift (hunk 1 above)
Comment (bmonk) [reply PhoneNumberKit/PartialFormatter.swift]: reformatted

**Comment by bmonk** [comment]
Closing because upstream resolved this issue via its PR https://github.com/marmelroy/PhoneNumberKit/pull/329

**Comment by bmonk** [comment]
The Simulator-only `CoreTelephony` logging issue will be resolved by our PR on upstream https://github.com/marmelroy/PhoneNumberKit/pull/331. 

I'm hopeful that merges RSN, but will open a PR here it doesn't. 

Either way, it won't affect [ZelleSDK PR 582](https://github.com/Banno/BannoZelleSDK-iOS/pull/582) for releases, because that branch sets a default Region code on `PhoneNumberKit`, and the `CoreTelephony ` logging issue only occurs when  no default is set 
*and* running on Simulator.

**Pull request closed without merging**